### PR TITLE
[com_fields] Rename alias to name and add a check if name is unique

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue # .
https://github.com/joomla/joomla-cms/issues/13335

### Summary of Changes
1. Changing the alias in name in database and code.
2. changed the position of the field -> position of field alias is empty
                                                       -> field name is shown with the options
3. add a check, if the field name is unique

### Testing Instructions
1. Change the value in the database for the com_fields alias to name

SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
SET time_zone = "+00:00";

ALTER TABLE `#__fields` CHANGE  `alias` `name` VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '';

(or make new installation of Joomla after applying this path).

2. Apply this patch

3. Create and edit Custom fields in the frontend and in the backend 

### Documentation Changes Required
No

### CODE REVIEW

File: administrator/components/com_fields/helpers/fields.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static function extract($contextString, $item = null)
 	 * Should the value being prepared to be shown in an HTML context then
 	 * prepareValue must be set to true. No further escaping needs to be done.
 	 * The values of the fields can be overridden by an associative array where the keys
-	 * has to be an id or an alias and it's corresponding value.
+	 * has to be a name and it's corresponding value.
```
Comment (brianteeman) [inline administrator/components/com_fields/helpers/fields.php]: i know this was there before but it should be "have" not "has"

File: administrator/language/en-GB/en-GB.ini [hunk 2]
```diff
@@ -180,6 +180,9 @@ JFIELD_ACCESS_LABEL="Access"
 JFIELD_ALIAS_DESC="The Alias will be used in the SEF URL. Leave this blank and Joomla will fill in a default value from the title. This value will depend on the SEO settings (Global Configuration->Site). <br />Using Unicode will produce UTF-8 aliases. You may also enter manually any UTF-8 character. Spaces and some forbidden characters will be changed to hyphens.<br />When using default transliteration it will produce an alias in lower case and with dashes instead of spaces. You may enter the Alias manually. Use lowercase letters and hyphens (-). No spaces or underscores are allowed. Default value will be a date and time if the title is typed in non-latin letters ."
 JFIELD_ALIAS_LABEL="Alias"
 JFIELD_ALIAS_PLACEHOLDER="Auto-generate from title"
+JFIELD_NAME_DESC="The name will be used for identifying the field. Joomla! will fill in a value computed from the title."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]: ==>
he name will be used to identify the field. Joomla will fill in a value computed from the title."

**Comment by astridx** [comment]
Done. Thanks. 
Is there a rule when we use the exclamation mark after the word Joomla?

**Comment by brianteeman** [comment]
https://developer.joomla.org/en-gb-user-interface-text-guidelines/joomla-name.html

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 164c8d226180de393c875bce094239ba3b7f9f50<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14554">issues.joomla.org/tracker/joomla-cms/14554</a>.</sub>

**Comment by laoneo** [comment]
I could save the field with a different name, edit an article and change the values and checked if the values do correctly appear on the front.

**Comment by laoneo** [comment]
You need to fix some code style errors to make drone happy http://213.160.72.75/joomla/joomla-cms/2006

**Comment by astridx** [comment]
@laoneo Thank you for testing :) 
I think drone is now happy: http://213.160.72.75/joomla/joomla-cms/2053 :)

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on d336fbfd6744dd9940fe6e8dbaf039af9bf7863f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14554">issues.joomla.org/tracker/joomla-cms/14554</a>.</sub>

**Comment by ysrostov** [comment]
Tested successfully with the fresh installation. I created two groups in Content menu and tried to create two custom fields in each group with similar name - checking for duplicate names is implemented and works fine.

**Comment by Bakual** [comment]
Setting RTC after two tests

**Comment by laoneo** [comment]
Thanks @astridx!

**Pull request merged**